### ISSUE TITLE
Allow Java projects to click on Ceylon declarations

### ISSUE THREAD

**Issue by FroMage** [ceylon/ceylon-ide-eclipse#1776]
Figure out if we can make Java projects that depend on Ceylon projects navigate to Ceylon types by clicking on them. ATM you get to the bytecode, which is not useful.

**Comment by gavinking** [comment]
Eyeballing the code, it's not at all clear why this doesn't already work...

**Comment by davidfestal** [comment]
yes, that's what I thought also. Probably something in `JavaSearch`. I'll have a look.

**Comment by gavinking** [comment]
In fact, it works for me. @FroMage you have to select Open Ceylon Declaration from the little drop down menu thingy, not just click.

**Comment by gavinking** [comment]
To be clear: it _only_ works if the Ceylon project exports its classpath containers.

**Comment by FroMage** [comment]
OK, so I found that popup. For it to become visible I need to `Ctrl-hover-move` the mouse over the declaration for a second before it shows up. Just Ctrl-clicking it leads to the bytecode. That's not enough, guys, nobody can discover that.

**Comment by FroMage** [comment]
Note that the tooltips could be improved too:

```
Open Declarationcom.foo.bar.CeylonType

@Ceylon
@SharedAnnotation$annotation$
Note: This element neither has attached source nor attached Javadoc and hence no Javadoc could be found.
```

**Comment by gavinking** [comment]
> For it to become visible I need to `Ctrl-hover-move` the mouse over the declaration for a second before it shows up. Just Ctrl-clicking it leads to the bytecode. That's not enough, guys, nobody can discover that.

Open an issue against Eclipse. On Mac the menu pops up *immediately* with no delay.

**Comment by gavinking** [comment]
> Note that the tooltips could be improved too:

What tooltips are these? I've never seen a tooltip like that.

**Comment by gavinking** [comment]
@maxandersen Do you know how to make our hyperlink detector the first one in the Java editor?

**Comment by maxandersen** [comment]
I do not think there are a way to sort/order hyperlink detectors.

But I also do not think this is the right way to solve this issue.

If you are ending up with bytecode in an editor it is because the source lookup mechanism is not right. As I recall it you can add custom source code lookup to projects which will be activated when eclipse can't locate it - @fbricon might know of the best place to add these.

**Comment by gavinking** [comment]
@maxandersen even if the source lookup was able to locate the Ceylon source code, that still wouldn't be good enough, since it would still not be enough to convince the Java editor to open the linked source code in the *Ceylon* editor instead of the Java editor.

**Comment by maxandersen** [comment]
do you know of a bug already open against JDT to fix that ? If not, lets get one open with this imo very valid usecase.

**Comment by gavinking** [comment]
I do not, no.
